homeassistant/components/trace/trace.py

```diff
@@ -0,0 +1,35 @@
+"""Support for automation and script tracing and debugging."""
+from homeassistant.core import callback
+
+from .const import DATA_TRACE
+
+
+@callback
+def get_debug_trace(hass, automation_id, run_id):
+    """Return a serializable debug trace."""
+    return hass.data[DATA_TRACE][automation_id][run_id]
+
+
+@callback
+def get_debug_traces_for_automation(hass, automation_id, summary=False):
```

Comment:
Why is this `_for_automation` ? I would expect a generic fetch method that also tells integration to pass in their `DOMAIN`.

Reply:
Fixed in #48288